packages/gasket-plugin-intl/lib/middleware.js

```diff
@@ -29,11 +29,22 @@ function formatLocale(language) {
 }
 
 module.exports = function middlewareHook(gasket) {
-  const { defaultLocale, basePath, localesMap, localesDir, manifestFilename, locales } = getIntlConfig(gasket);
+  const {
+    defaultLocale,
+    basePath,
+    localesMap,
+    localesDir,
+    manifestFilename,
+    locales,
+    preloadLocales
+  } = getIntlConfig(gasket);
 
   const manifest = require(path.join(localesDir, manifestFilename));
   const localesParentDir = path.dirname(localesDir);
   const localeUtils = new LocaleUtils({ manifest, basePath });
+  if (preloadLocales) {
+    Object.keys(manifest.paths).map((localePath) => require(path.join(localesDir.slice(0, -8), localePath)));
```

Comment:
Using slice like this is risky, as `localesDir` can be configured to something else. Instead, you'd want to use `path.dirname` to get the parent directory, which `localesParentDir` has already been assigned.
```suggestion
    Object.keys(manifest.paths).map((localePath) => require(path.join(localesParentDir, localePath)));
```

Reply:
Ohh good point 👍. 
Using `localesParentDir` now, Thanks @agerard-godaddy